popup_offer/src/Plugin/Commerce/PromotionOffer/PopupOffer.php

```diff
@@ -1,293 +0,0 @@
-    $this->userData = $userData;
-    $this->currentUser = $currentUser;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition)
-  {
-    return new static(
-        $configuration,
-        $plugin_id,
-        $plugin_definition,
-        $container->get('commerce_price.rounder'),
-        $container->get('commerce_order.price_splitter'),
-        $container->get('plugin.manager.commerce_condition'),
-        $container->get('entity_type.manager'),
-        $container->get('user.data'),
-        $container->get('current_user')
-    );
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function defaultConfiguration()
-  {
-    return [
-            'main_product' => '0',
-            'discount_product1' => '0',
-            'discount_product2' => '0',
-            'discount_product3' => '0',
-            'discount_product4' => '0',
-            'discount_value' => 0,
-        ] + parent::defaultConfiguration();
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function buildConfigurationForm(array $form, FormStateInterface $form_state)
-  {
-    $form += parent::buildConfigurationForm($form, $form_state);
-
-    // Remove the main fieldset.
-    $form['#type'] = 'container';
-
-    $form['product'] = [
-        '#type' => 'fieldset',
-        '#title' => $this->t('Target product'),
-        '#collapsible' => FALSE,
-    ];
-    $form['product']['name'] = [
-        '#type' => 'entity_autocomplete',
-        '#title' => $this->t('Product name'),
-        '#default_value' => $this->getProductNameByID([$this->configuration['main_product']]),
-        '#type' => 'entity_autocomplete',
-        '#target_type' => 'commerce_product',
-    ];
-
-    $form['discount'] = [
-        '#type' => 'fieldset',
-        '#title' => $this->t('Discount products'),
-        '#collapsible' => FALSE,
-    ];
-    $form['discount']['product1'] = [
-        '#type' => 'entity_autocomplete',
-        '#title' => $this->t('Name of discount product'),
-        '#default_value' => $this->getProductNameByID([$this->configuration['discount_product1']]),
-        '#type' => 'entity_autocomplete',
-        '#target_type' => 'commerce_product',
-    ];
-    $form['discount']['product2'] = [
-        '#type' => 'entity_autocomplete',
-        '#title' => $this->t('Name of discrount product'),
-        '#default_value' => $this->getProductNameByID([$this->configuration['discount_product2']]),
-        '#type' => 'entity_autocomplete',
-        '#target_type' => 'commerce_product',
-    ];
-    $form['discount']['product3'] = [
-        '#type' => 'entity_autocomplete',
-        '#title' => $this->t('Name of discount product'),
-        '#default_value' => $this->getProductNameByID([$this->configuration['discount_product3']]),
-        '#type' => 'entity_autocomplete',
-        '#target_type' => 'commerce_product',
-    ];
-    $form['discount']['product4'] = [
```

Comment:
foreach (range(1, 4) as $key) {

would be much better than product_1, product_2 etc

Reply:
Agree.